libraries/AP_Follow/AP_Follow.cpp

```diff
@@ -340,19 +351,45 @@ bool AP_Follow::handle_global_position_int_message(const mavlink_message_t &msg)
         if ((packet.lat == 0 && packet.lon == 0)) {
             return false;
         }
+        // ignore message if we already received a more recent one
+        if(_target_location_last_time_boot_ms != 0 && packet.time_boot_ms <= _target_location_last_time_boot_ms ) {
+            return false;
+        }
+        _target_location_last_time_boot_ms = packet.time_boot_ms;
 
         _target_location.lat = packet.lat;
         _target_location.lng = packet.lon;
 
         // select altitude source based on FOLL_ALT_TYPE param 
+#if APM_BUILD_TYPE(APM_BUILD_ArduPlane)
+        switch(_alt_type) {
+            case AP_FOLLOW_ALT_TYPE_DEFAULT:
+                _target_location.set_alt_cm(packet.alt * 0.1, Location::AltFrame::ABSOLUTE);
+                break;
+            case AP_FOLLOW_ALTITUDE_TYPE_RELATIVE:
+            case AP_FOLLOW_ALTITUDE_TYPE_ORIGIN:
```

Comment:
Remove origin case?

Reply:
As per above - this means the ordinal values match Location:AltFrame, which is kind of useful.

Reply:
Sure, but it doesn't do relative to origin, it should return false out of the function as a invalid setting. Or it should come under the a default case in the switch. Adding a enum value for origin but then using it for relative home is just confusing.